src/hooks/useFriction.ts

```diff
@@ -23,42 +25,72 @@ export const useFriction = <M extends HTMLElement>({
    */
   const ref = React.useRef<M>(null);
 
+  /**
+   * Store a ref to the controller. This will allow a user to
+   * start and stop animations at will.
+   */
+
   const { controller } = React.useMemo(() => {
-    const { interpolator, property, values } = getInterpolatorForPair({
-      from,
-      to,
-    });
+    const interpolators = getInterpolatorsForPairs({ from, to });
 
     return friction1D({
       config,
       onUpdate: ({ position }) => {
-        const value = interpolator({
-          range: [
-            0,
-            getMaxDistanceFriction({
-              mu: config.mu,
-              initialVelocity: config.initialVelocity,
-            }),
-          ],
-          domain: [values.from, values.to],
-          value: position[0],
-        });
+        interpolators.forEach(({ interpolator, property, values }) => {
+          const value = interpolator({
+            range: [
+              0,
+              getMaxDistanceFriction({
+                mu: config.mu,
+                initialVelocity: config.initialVelocity * 1000,
+              }),
+            ],
+            domain: [values.from, values.to],
+            value: position[0],
+          });
 
-        if (ref.current) {
-          ref.current.style[property as any] = `${value}`;
-        }
+          if (ref.current) {
+            ref.current.style[property as any] = `${value}`;
+          }
+        });
       },
       onComplete: () => {
         /**
          * Ensure our animation has reached the to value when the physics stopping
          * condition has been reached.
          */
-        if (ref.current && ref.current.style[property as any] !== values.to) {
-          ref.current.style[property as any] = values.to;
-        }
+        interpolators.forEach(({ property, values }) => {
+          if (ref.current && ref.current.style[property as any] !== values.to) {
+            ref.current.style[property as any] = values.to;
+          }
+        });
       },
     });
   }, [from, to, config]);
 
-  return [{ ref }, controller];
+  const controllerRef = React.useRef<Controller>({
+    start: controller.start,
+    stop: () => {},
+  });
+
+  React.useLayoutEffect(() => {
+    if (immediate && !delay) {
+      const { stop } = controller.start();
+      controllerRef.current.stop = stop;
+    }
+
+    let timerId: number;
+    if (immediate && delay) {
+      timerId = setTimeout(() => {
+        const { stop } = controller.start();
+        controllerRef.current.stop = stop;
+      }, delay);
+    }
+
+    return () => {
+      timerId && clearTimeout(timerId);
+    };
+  }, [immediate, delay, controller]);
```

Comment:
`useLayoutEffect` is important here. Once we call `controller.start` we're going to want to synchronously mutate the DOM via the `ref` here. We want this mutation to take effect before the browser actually paints, hence why we use `useLayoutEffect` here.